src/ClassDiscovery.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace Http\Discovery;
+
+/**
+ * Registry that based find results on class existence
+ *
+ * @author David de Boer <david@ddeboer.nl>
+ */
+abstract class ClassDiscovery
+{
+    /**
+     * Cached class
+     *
+     * @var object
+     */
+    protected static $cache;
+
+    /**
+     * Conditions and classes
+     *
+     * @var array
+     */
+    protected static $classes = [];
+
+    /**
+     * Register a factory
+     *
+     * @param string $name
+     * @param string $condition
+     * @param string $class
+     */
+    public static function register($name, $condition, $class = null)
```

Comment:
To answer your question: two possible way I can see: make condition required argument or check it in cases when required.

For the long term, i would convert the condition to something that can be evaluated as boolean. So we could accept a callable or boolean as well. In that case, null is obviously false.